core_impl/src/test/java/io/opencensus/stats/StatsTestUtil.java

```diff
@@ -53,108 +71,226 @@ private static StatsContextImpl createContext(
+            if (max == Double.NEGATIVE_INFINITY && min == Double.POSITIVE_INFINITY) {
+              assertThat(arg.getMax()).isNegativeInfinity();
+              assertThat(arg.getMin()).isPositiveInfinity();
+            } else {
+              assertThat(arg.getMax()).isWithin(tolerance).of(max);
+              assertThat(arg.getMin()).isWithin(tolerance).of(min);
+            }
+            return null;
+          }
+        },
+        new Function<MutableMean, Void>() {
+          @Override
+          public Void apply(MutableMean arg) {
+            assertThat(arg.getMean()).isWithin(tolerance).of(mean);
+            return null;
+          }
+        },
+        new Function<MutableStdDev, Void>() {
+          @Override
+          public Void apply(MutableStdDev arg) {
+            assertThat(arg.getStdDev()).isWithin(tolerance).of(stdDev);
+            return null;
+          }
+        });
   }
 
   /**
-   * Asserts that the two sets of {@code DistributionAggregate}s are equivalent, with a given
-   * tolerance. The tolerance is used when comparing the mean and sum of values. The order of the
-   * {@code DistributionAggregate}s has no effect. The expected parameter is last, because it is
-   * likely to be a larger expression.
+   * Compare the {@code AggregationData} with expected values based on its underlying type. The
+   * expected values are all optional (nullable).
    *
+   * @param aggregate the {@code AggregationData} to be verified.
+   * @param sum the expected sum value, if aggregation is a {@code SumData}.
+   * @param count the expected count value, if aggregation is a {@code CountData}.
+   * @param bucketCounts the expected bucket counts, if aggregation is a {@code HistogramData}.
+   * @param min the expected min value, if aggregation is a {@code RangeData}.
+   * @param max the expected max value, if aggregation is a {@code RangeData}.
+   * @param mean the expected mean value, if aggregation is a {@code MeanData}.
+   * @param stdDev the expected standard deviation, if aggregation is a {@code StdDevData}.
    * @param tolerance the tolerance used for {@code double} comparison.
-   * @param actual the actual test result.
-   * @param expected the expected value.
-   * @throws AssertionError if the {@code DistributionAggregate}s don't match.
+   * @throws AssertionError if {@code AggregationData} doesn't match with expected value.
    */
-  static void assertDistributionAggregatesEquivalent(
-      double tolerance,
-      Collection<DistributionAggregate> actual,
-      Collection<DistributionAggregate> expected) {
-    Function<DistributionAggregate, List<Tag>> getTagsFunction =
-        new Function<DistributionAggregate, List<Tag>>() {
+  static void assertAggregationDataEquals(
+      AggregationData aggregate, final Double sum, final Long count, final List<Long> bucketCounts,
+      final Double min, final Double max, final Double mean, final Double stdDev,
+      final double tolerance) {
```

Comment:
I would prefer to keep this method here since I'll use it for comparing the `AggregationMap`: https://github.com/census-instrumentation/opencensus-java/pull/493/commits/72ed9e339f9aade1ba02565b19f5799f4bfbf33b#diff-c24b66b69d113e9ee0e6ddcaea2cf739R211